src/internal/connector/onedrive/restore.go

```diff
@@ -223,62 +261,74 @@ func RestoreCollection(
 
 		case itemData, ok := <-items:
 			if !ok {
-				return metrics, nil
+				complete = true
+				break
```

Comment:
Since it's a little difficult to parse with a select in a for-loop could you please add a comment that this will get us out of the select and then the check above will get us out of the for-loop